pennylane_lightning/core/src/simulators/lightning_gpu/StateVectorCudaManaged.hpp

```diff
@@ -359,33 +360,128 @@ class StateVectorCudaManaged
      * @param tgt_wires Wires to apply gate to.
      * @param adjoint Indicates whether to use adjoint of gate.
      * @param params Optional parameter list for parametric gates.
-     * @param gate_matrix Optional std gate matrix if opName doesn't exist.
+     * @param gate_matrix Optional gate matrix vector on the host if opName
+     * doesn't exist.
      */
-    void applyOperation(
-        const std::string &opName,
-        const std::vector<std::size_t> &controlled_wires,
-        const std::vector<bool> &controlled_values,
-        const std::vector<std::size_t> &tgt_wires, bool adjoint = false,
-        const std::vector<Precision> &params = {0.0},
-        [[maybe_unused]] const std::vector<ComplexT> &gate_matrix = {}) {
-        PL_ABORT_IF_NOT(opName == "GlobalPhase",
-                        "Only GlobalPhase gate is supported.");
+    void applyOperation(const std::string &opName,
+                        const std::vector<std::size_t> &controlled_wires,
+                        const std::vector<bool> &controlled_values,
+                        const std::vector<std::size_t> &tgt_wires,
+                        bool adjoint = false,
+                        const std::vector<Precision> &params = {0.0},
+                        const std::vector<ComplexT> &gate_matrix = {}) {
```

Comment:
For consistency with the `applyOperation` method with 5 arguments, should we accept `std::vector<CFP_t>` for the matrix here, and have a separate method that accepts `std::vector<ComplexT>` which does the `complextoCu` ?

Reply:
Good one. We can revisit this in a follow-up PR